Core.DomainModel/Organization/Organization.cs

```diff
@@ -198,27 +198,17 @@ public Maybe<UIModuleCustomization> GetUiModuleCustomization(string module)
         {
             if (root == null) throw new ArgumentNullException(nameof(root));
 
-            IExternalOrganizationalHierarchyUpdateStrategy strategy;
-            //Pre-validate
-            switch (origin)
-            {
-                case OrganizationUnitOrigin.STS_Organisation:
-                    if (StsOrganizationConnection?.Connected != true)
-                    {
-                        return new OperationError($"Not connected to {origin:G}. Please connect before performing an update", OperationFailure.Conflict);
-                    }
-                    strategy = StsOrganizationConnection.GetUpdateStrategy();
-                    break;
-                case OrganizationUnitOrigin.Kitos:
-                    return new OperationError("Kitos is not an external source", OperationFailure.BadInput);
-                default:
-                    throw new ArgumentOutOfRangeException();
-            }
+            return GetStsOrganizationConnection(origin)
+                .Bind<OrganizationTreeUpdateConsequences>(connection =>
```

Comment:
you have the connection in your parameter yet you still access the property. Had you extracted an abstraction you would have called the param and that would actually improve the design.. also a good place to visit in our talk